shenyu-proxy/proxy-remote/src/main/java/org/apache/shenyu/proxy/remote/netty/NettyServerHandler.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.proxy.remote.netty;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.timeout.IdleState;
+import io.netty.handler.timeout.IdleStateEvent;
+
+/**
+ * NettyServerHandler .
+ * Processing of netty server.
+ *
+ * @author sixh chenbin
+ */
+@ChannelHandler.Sharable
+public class NettyServerHandler extends ChannelDuplexHandler {
+    
+    private final org.apache.shenyu.proxy.remote.ChannelHandler channelHandler;
+    
+    public NettyServerHandler(final org.apache.shenyu.proxy.remote.ChannelHandler channelHandler) {
+        this.channelHandler = channelHandler;
+    }
+    
+    @Override
+    public void channelActive(final ChannelHandlerContext ctx) throws Exception {
+        super.channelActive(ctx);
+        NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel());
+        try {
+            channelHandler.connection(channel);
+        } finally {
+            NettyChannel.removeChannel(ctx.channel());
+        }
+    }
+    
+    @Override
+    public void channelInactive(final ChannelHandlerContext ctx) throws Exception {
+        super.channelInactive(ctx);
+        try {
+            NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel());
+            channelHandler.disConnection(channel);
+        } finally {
+            NettyChannel.removeChannel(ctx.channel());
+        }
+    }
+    
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel());
+        try {
+            channelHandler.receive(channel, msg);
+        } finally {
+            NettyChannel.removeChannel(ctx.channel());
+        }
+    }
+    
+    @Override
+    public void write(final ChannelHandlerContext ctx,
+                      final Object msg, final ChannelPromise promise) throws Exception {
+        super.write(ctx, msg, promise);
+        NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel());
+        try {
+            channelHandler.sent(channel, msg);
+        } finally {
+            NettyChannel.removeChannel(ctx.channel());
+        }
+    }
+    
+    @Override
+    public void userEventTriggered(final ChannelHandlerContext ctx, final Object evt) throws Exception {
+        if (evt instanceof IdleStateEvent) {
+            IdleState state = ((IdleStateEvent) evt).state();
+            NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel());
+            try {
+                switch (state) {
+                    case READER_IDLE:
+                    case WRITER_IDLE:
+                    case ALL_IDLE:
+                        channel.close();
```

Comment:
why here we have three status listed , and constructor only have readerIdleTimeSeconds set, can WRITER_IDLE trigged?